ca.go

```diff
@@ -18,19 +20,29 @@ type CA struct {
 	// may be set to something like `nil`.
 	Rand io.Reader
 
-	// Wrapped `crypto.Signer` to preform OpenSSH CA operations.
-	Signer ssh.Signer
+	// Function to choose which signer to use based on request information
+	ChooseSigner func(events.APIGatewayProxyRequest) (crypto.Signer, error)
```

Comment:
I'd prefer if it instead of taking a func, this took an interface. That makes it much clearer to have state on the signerchooser.

An example of this is that the default one would be able to look up the env vars once on creation, and no on every signature.